cmds/core/shutdown/shutdown_linux.go

```diff
@@ -41,48 +41,54 @@ var (
 		"suspend": unix.LINUX_REBOOT_CMD_SW_SUSPEND,
 		"-s":      unix.LINUX_REBOOT_CMD_SW_SUSPEND,
 	}
-	reboot = unix.Reboot
-	delay  = time.Sleep
 )
 
 func usage() {
-	log.Fatalf("shutdown [<-h|-r|-s|halt|reboot|suspend> [time [message...]]]")
+	fmt.Println("shutdown [<-h|-r|-s|halt|reboot|suspend> [time [message...]]]")
 }
 
-func main() {
-	a := os.Args
-	if len(a) == 1 {
-		a = append(a, "halt")
+func shutdown(args []string) (uint, error) {
```

Comment:
I think this would be more readable if you were not passing the program name in :-), and since you allow it to have 0 args: 
func shutdown(args ...string) (uint, error) {